src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx

```diff
@@ -1,127 +1,172 @@
-            alignItems: 'center',
-            columnGap: 4
-        },
-        modelNameText: {
-            margin: 0
-        },
-        dot: {
-            width: 8,
-            height: 8,
-            borderRadius: 4
-        },
-        // Property side
-        checkboxContainer: {
-            display: 'flex',
-            flexDirection: 'column',
-            rowGap: 8
-        }
-    });
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [properties, _setProperties] = useState<IModelProperty[]>(
+        dataManagementContextState.modifiedAssets.properties
+    );
+    const [models, setModels] = useState<IModelExtended[]>(
+        buildViewModel(
+            dataManagementContextState.initialAssets.models,
+            dataManagementContextState.modifiedAssets.models
+        )
+    );
+
+    useEffect(() => {
+        // onUnmount update assets
+        return () => {
+            dataManagementContextDispatch({
+                type: DataManagementContextActionType.UPDATE_MODIFIED_ASSETS,
+                payload: {
+                    data: {
+                        ...dataManagementContextState.modifiedAssets,
+                        models,
+                        properties
+                    }
+                }
+            });
+        };
+    }, [
+        dataManagementContextDispatch,
+        dataManagementContextState.modifiedAssets,
+        models,
+        properties
+    ]);
+
+    // Styles
+    const style = useMemo(() => getModelListsStyles(palette), [palette]);
 
     //callbacks
     const handleOnCheck = useCallback(
         (modelId: string, propertyId: string, checked: boolean) => {
-            wizardNavigationContextDispatch({
-                type:
-                    WizardNavigationContextActionType.SET_MODEL_PROPERTY_SELECTED,
-                payload: { modelId, propertyId, checked }
-            });
+            const tempModels = Array.from(models);
+            const modelToModifyIdx = tempModels.findIndex(
+                (m) => m.id === modelId
+            );
+            if (modelToModifyIdx >= 0) {
+                if (checked) {
+                    tempModels[modelToModifyIdx].propertyIds.push(propertyId);
+                } else {
+                    tempModels[modelToModifyIdx].propertyIds.filter(
+                        (p) => p !== propertyId
+                    );
+                }
+            }
+            setModels(tempModels);
         },
-        [wizardNavigationContextDispatch]
+        [models]
+    );
+
+    const handleModelNameChange = useCallback(
+        (modelId: string, modelName: string) => {
+            const tempModels = Array.from(models);
+            const modelToModifyIdx = tempModels.findIndex(
+                (m) => m.id === modelId
+            );
+            if (modelToModifyIdx >= 0) {
+                tempModels[modelToModifyIdx].name = modelName;
+            }
+            setModels(tempModels);
```

Comment:
same here: So here shouldn't we also update the modifiedAssets in DataManagementContext? or is it done in "next/finish" button in TwinVerificationStep - I didn't see that there either?